src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/Serialization/ProjectSnapshotHandleJsonConverter.cs

```diff
@@ -49,21 +47,21 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                     case nameof(ProjectSnapshotHandle.RootNamespace):
                         if (reader.Read())
                         {
-                            rootNamespace = (string)reader.Value;
+                            rootNamespace = (string)reader.Value!;
                         }
 
                         break;
                 }
 
                 return (reader, objectType, existingValue, serializer, filePath, configuration, rootNamespace);
-            }, (reader, objectType, existingValue, serializer, filePath: (string)null, configuration: (RazorConfiguration)null, rootNamespace: (string)null));
+            }, (reader, objectType, existingValue, serializer, filePath: null as string, configuration: null as RazorConfiguration, rootNamespace: null as string));
 
-            return new ProjectSnapshotHandle(filePath, configuration, rootNamespace);
+            return new ProjectSnapshotHandle(filePath!, configuration, rootNamespace);
         }
 
-        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        public override void WriteJson(JsonWriter writer, object? value, JsonSerializer serializer)
         {
-            var handle = (ProjectSnapshotHandle)value;
+            var handle = (ProjectSnapshotHandle)value!;
```

Comment:
Not sure about this one. It didn't have any guard and my initial goal was to just enable `nullable` without changing the logic. So asking you: should we make any guard here, e.g. a `throw`?

Reply:
I personally don't have a strong opinion about it, I think the current form looks fine to me.